django/contrib/auth/__init__.py

```diff
@@ -121,16 +121,22 @@ def login(request, user, backend=None):
     have to reauthenticate on every request. Note that data set during
     the anonymous session is retained when the user logs in.
     """
-    session_auth_hash = ''
+    session_auth_hash_sha256 = ''
+    session_auth_hash_sha1 = ''
     if user is None:
         user = request.user
     if hasattr(user, 'get_session_auth_hash'):
-        session_auth_hash = user.get_session_auth_hash()
```

Comment:
May be introducing an another two functions which are undocumented is fine?
`get_session_auth_hash_sha1` and `get_session_auth_hash_sha256` without changing the behavior of `get_session_auth_hash()`?